### PR TITLE
Move cache out of /srv/mediawiki/w directory

### PR DESCRIPTION
**Pull request by Universal-Omega**
Per [T7192](https://phabricator.miraheze.org/T7192)

### CODE REVIEW

File: LocalSettings.php
Comment (Southparkfan) [inline LocalSettings.php:777]: 1) Requires a puppet patch to create a writable directory
2) Please do not merge this _before_ denying access to this new folder: https://github.com/miraheze/puppet/blob/master/modules/mediawiki/templates/mediawiki-includes.conf.erb#L11-L13
3) Per the https://en.wikipedia.org/wiki/Filesystem_Hierarchy_Standard, /var/cache may be the better location?

**Review by Southparkfan — CHANGES_REQUESTED** [review]

File: LocalSettings.php
Comment (Universal-Omega) [reply LocalSettings.php:777]: 1. yep that's why I marked this one as draft for now so it doesn't get merged before that's done.
2. Oh, right, thanks!
3. That actually may be better, yeah. Thanks!

**Comment by paladox** [comment]
We could do /srv/mediawiki/cache (where i18n & MW go).

So /srv/mediawiki/cache/i18n & /srv/mediawiki/managewiki.

**Comment by JohnFLewis** [comment]
> That actually may be better, yeah.

If you think /var/cache as SPF suggested is better, why did you revert that change?

**Comment by Universal-Omega** [comment]
> > That actually may be better, yeah.
> 
> If you think /var/cache as SPF suggested is better, why did you revert that change?

Because I'm not exactly sure what to do. In all honesty after more thought /srv/mediawiki/cache does make more sense here, but I also still want to take SPFs advice here. And there's been some conflicting recommendations and I'm not really sure what to do here.

**Comment by Universal-Omega** [comment]
Puppet PR: miraheze/puppet#1772

**Comment by Reception123** [comment]
Seems fine to me, though would like @The-Voidwalker and/or @Southparkfan to have a final look at this and puppet before we make the move

**Comment by The-Voidwalker** [comment]
Approved from a code perspective. I think everything that would need to be changed was updated. Regarding `/srv/mediawiki/cache` vs `/var/cache/mediawiki`, I don't really have much to say. It makes sense to me to keep all the mediawiki stuff together, and I'd approve as-is unless there's good reason or precedent (in our infrastructure) to do otherwise.

**Pull request merged**
